src/proxy/IReplicatedMap.ts

```diff
@@ -0,0 +1,169 @@
+import * as Promise from 'bluebird';
+import {DistributedObject} from '../DistributedObject';
+import {Predicate} from '../core/Predicate';
+import {IMapListener} from '../core/MapListener';
+import Long = require('long');
+import {ArrayComparator} from '../util/ArrayComparator';
+
+export interface IReplicatedMap<K, V> extends DistributedObject {
+    /**
+     * Associates a given value to the specified key and replicates it to the
+     * cluster. If there is an old value, it will be replaced by the specified
+     * one and returned from the call.
+     *
+     * @param key key with which the specified value is to be associated.
+     * @param value value to be associated with the specified key.
+     * @param ttl milliseconds to be associated with the specified key-value pair.
+     * @return old value if there was any, `null` otherwise.
+     */
+    put(key: K, value: V, ttl: Long|number|string): Promise<V>;
```

Comment:
I still argue against accepting string as a valid argument. If we do this, then it is our responsibility to convert this string to a valid number. Users can provide too big numbers or malformed inputs as strings. This conversion is out of this client's interest. Even if I accept that, I do not see the relevant code that does the verification and conversion in case of string `ttl` argument. If we put that code here, we need to maintain it. Also, accepting a string in `ttl` area does not seem semantically correct to me. `ttl` is semantically a number(actually a non-negative integer but JS allows us to restrict it only to the point of a number). I see this as more of a burden to maintainers than it is a convenience to users.